test/spec/modules/magniteAnalyticsAdapter_spec.js

```diff
@@ -2039,4 +2059,117 @@ describe('magnite analytics adapter', function () {
+        }
+      });
+      config.setConfig({ rubicon: { updatePageView: true } });
+    });
+
+    it('should add a no-bid bid to the add unit if it recieves one from the server', () => {
+      const bidResponse = utils.deepClone(MOCK.NO_BID);
+      const auctionInit = utils.deepClone(MOCK.AUCTION_INIT);
+
+      delete auctionInit.adUnits[0].sizes;
+
+      bidResponse.requestId = 'fakeId';
+      events.emit(AUCTION_INIT, auctionInit);
+      events.emit(BID_REQUESTED, MOCK.BID_REQUESTED);
+      events.emit(BID_RESPONSE, bidResponse)
+      events.emit(BIDDER_DONE, MOCK.BIDDER_DONE);
+      events.emit(AUCTION_END, MOCK.AUCTION_END);
+      clock.tick(rubiConf.analyticsBatchTimeout + 1000);
+
+      let message = JSON.parse(server.requests[0].requestBody);
+      expect(utils.generateUUID.called).to.equal(true);
+
+      expect(message.auctions[0].adUnits[0].bids[1]).to.deep.equal(
+        {
+          bidder: 'mgnipbs',
+          source: 'server',
+          status: 'no-bid',
+          bidResponse: {
+            "bidPriceUSD": 0,
+            "conversionError": true
+          },
+          oldBidId: 'fakeId',
+          unknownBid: true,
+          bidId: 'fakeId',
+          clientLatencyMillis: -139101369960
+        }
+      );
+    });
+  });
+
+  describe('SEAT_NON_BID events', () => {
+    let seatnonbid;
+
+    const runNonBidAuction = () => {
+      events.emit(AUCTION_INIT, MOCK.AUCTION_INIT);
+      events.emit(BID_REQUESTED, MOCK.BID_REQUESTED);
+      events.emit(SEAT_NON_BID, seatnonbid)
+      events.emit(BIDDER_DONE, MOCK.BIDDER_DONE);
+      events.emit(AUCTION_END, MOCK.AUCTION_END);
+      clock.tick(rubiConf.analyticsBatchTimeout + 1000);
+    };
+    const checkStatusAgainstCode = (status, code, error, index) => {
+      seatnonbid.seatnonbid[0].nonbid[0].status = code;
+      runNonBidAuction();
+      let message = JSON.parse(server.requests[index].requestBody);
+      let bid = message.auctions[0].adUnits[0].bids[1];
+
+      if (error) {
+        expect(bid.error).to.deep.equal(error);
+      } else {
+        expect(bid.error).to.equal(undefined);
+      }
+      expect(bid.source).to.equal('server');
+      expect(bid.status).to.equal(status);
+      expect(bid.isSeatNonBid).to.equal(true);
+    };
+    beforeEach(() => {
+      magniteAdapter.enableAnalytics({
+        options: {
+          endpoint: '//localhost:9999/event',
+          accountId: 1001
+        }
+      });
+      config.setConfig({ rubicon: { updatePageView: true } });
+      seatnonbid = utils.deepClone(MOCK.SEAT_NON_BID);
+    });
+
+    it('adds seatnonbid info to bids array', () => {
+      runNonBidAuction();
+      let message = JSON.parse(server.requests[0].requestBody);
+
+      expect(message.auctions[0].adUnits[0].bids[1]).to.deep.equal(
+        {
+          bidder: 'rubicon',
+          source: 'server',
+          status: 'no-bid',
+          isSeatNonBid: true,
+          clientLatencyMillis: -139101369960
+        }
```

Comment:
So this one tests if the status is not in the status map, and then below test is all of the other ones cool.